### PR TITLE
Add `localizedName` and `localizedDescription` to the admin Program form

### PR DESCRIPTION
**Pull request by carolinedanzi**
### Description
1. The non-block information for a program now contains: `adminName`, `adminDescription`, `localizedName`, and `localizedDescription`
2. When an admin edits a program, the `adminName` is not an editable field - we use this name internally to find versions of programs
3. Up next: actually provide a way for admins to enter localized names and descriptions for supported languages

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
#777 

Screen Shot 2021-04-20 at 4 44 09 PM

### CODE REVIEW

File: universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java [hunk 1]
```diff
@@ -222,16 +223,53 @@ public abstract Builder setLocalizedDescription(
 
     public abstract ProgramDefinition build();
 
+    /**
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java:226]: praise: thanks for adding the docstring!

File: universal-application-tool-0.0.1/app/services/program/ProgramService.java [hunk 2]
```diff
@@ -54,26 +55,38 @@
...
+   * @param adminName a name for this program
+   * @param adminDescription the description of what the program provides
    * @return the {@link ProgramDefinition} that was created if succeeded, or a set of errors if
    *     failed
    */
   ErrorAnd<ProgramDefinition, CiviFormError> createProgramDefinition(
-      String name, String description);
+      String adminName,
+      String adminDescription,
+      String defaultDisplayName,
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/services/program/ProgramService.java:66]: Suggestion (blocking): would you mind adding this to doctring?

**Review by hmhsu — APPROVED** [review]
Nice work!

File: universal-application-tool-0.0.1/app/services/program/ProgramService.java (hunk 2 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/services/program/ProgramService.java:69]: Great catch! Added

File: universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java [hunk 3]
```diff
@@ -127,15 +143,11 @@ public ProgramDefinition getProgramDefinition(long id) throws ProgramNotFoundExc
             .join());
   }
 
-  private ImmutableSet<CiviFormError> validateProgramDefinition(String name, String description) {
-    ImmutableSet.Builder<CiviFormError> errors = ImmutableSet.<CiviFormError>builder();
-    if (name.isBlank()) {
-      errors.add(CiviFormError.of("program name cannot be blank"));
+  private ImmutableSet<CiviFormError> validateProgramText(String... text) {
+    if (Arrays.stream(text).anyMatch(String::isBlank)) {
+      return ImmutableSet.of(CiviFormError.of("program information cannot be blank"));
```
[the remark this answers is not in the document]
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java]: Sure! I updated to include specific errors for these

**Pull request merged**